app/src/main/java/com/quipper/android/apollofrontpage/PostListFragment.kt

```diff
@@ -53,7 +44,26 @@ class PostListFragment : Fragment(), PostListAdapter.PostListHandler {
         }
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.postsData.observe(viewLifecycleOwner, Observer {
+            it ?: return@Observer
+            postListAdapter.submitList(it)
+        })
+
+        viewModel.errorData.observe(viewLifecycleOwner, Observer {
+            it ?: return@Observer
+            Toast.makeText(
+                requireContext(),
+                resources.getString(R.string.common_error),
+                Toast.LENGTH_SHORT
+            ).show()
+        })
+
+    }
+
     override fun handle(details: PostDetails) {
-        viewModel.increaseVote(details.id)
+        viewModel.upVote(details.id)
```

Comment:
[Minor] Should be `upvote`

Reply:
It will show typo warning of AS so let me keep as it is.